api/src/opentrons/protocol_runner/protocol_runner.py

```
@@ -156,9 +157,17 @@ async def run(
     def _load_json(self, protocol_source: ProtocolSource) -> None:
         protocol = self._json_file_reader.read(protocol_source)
         commands = self._json_translator.translate_commands(protocol)
-        liquids = self._json_translator.translate_liquids(protocol)
-        for liquid in liquids:
-            self._protocol_engine.add_liquid(liquid=liquid)
+
+        if not feature_flags.enable_load_liquid() and any(
+            c.commandType != "loadLiquid" for c in commands
+        ):
+            raise CommandDoesNotExistError("loadLiquid command is not yet supported.")
+
+        if feature_flags.enable_load_liquid():
+            liquids = self._json_translator.translate_liquids(protocol)
```

Should I raise an error here as well?

---

I don't think you need to raise an error here, right?

---

@Laura-Danielle I think I agree with you. we can load liquids but the load command shouldn't be allowed. @SyntaxColoring ?

---

If I understand correctly, I agree that this is good as written and that you shouldn't add an error here.

The JSON protocol schema allows extra fields, I believe, which the server will silently ignore. That should be the behavior that we preserve when the feature flag is disabled.